### PR TITLE
Improve protections and other random changes

### LINKED ISSUE #1801: [Bug]: Joined a public session where i saw spectate protection notification endless loop
### Describe the bug

Joined a public session
Saw Protection notification loop endless
Tried unloading and reloading the dll back (same notifcations start emerging again no effect)
image

### Steps To Reproduce

I am in that session right now. It is active.

### Expected Behavior

spectate protection should work correctly

### Logs and/or screenshots

[cout.log](https://github.com/YimMenu/YimMenu/files/12136830/cout.log)

### Store

Epic Games

### Additional context

_No response_

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### Coload

- [X] I've tested this without co-loading anything.

### LINKED ISSUE #1802: [Bug]: Spoofing settings are naved saved to the settings JSON.
### Describe the bug

Title.

### Steps To Reproduce

Change spoofing settings, close yim, reopen yim, settings are reset.

### Expected Behavior

Should save.

### Logs and/or screenshots

_No response_

### Store

Rockstar

### Additional context

_No response_

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### Coload

- [X] I've tested this without co-loading anything.

### PR DESCRIPTION
**Pull request by maybegreat48**
* Improved OOM kick (currently not blocked by most menus)
* Improved sync logging
* Improved protections

Fixes #1802 
Fixes #1801

### CODE REVIEW

**Comment by ghost** [comment]
Can you add g.reactions.modder_detection.process(player); to session.hpp inside the add_infraction() function? it's just that they don't want to merge my PR. This will close #1676

**Comment by maybegreat48** [comment]
> it's just that they don't want to merge my PR. This will close https://github.com/YimMenu/YimMenu/issues/1676

What makes you think that? You've never opened a PR in this repository as far as I can tell

**Comment by gir489returns** [comment]
> > it's just that they don't want to merge my PR. This will close #1676
> 
> What makes you think that? You've never opened a PR in this repository as far as I can tell

He's opened #1772, they are approved but not merged, probably because Yim hasn't gotten to it yet, like this dude expects Yim to live his life by this project.

File: src/hooks/misc/log_message_error_box.cpp [hunk 1]
```diff
@@ -31,10 +31,10 @@ namespace big
 
 	void hooks::log_error_message_box(rage::joaat_t joaated_error_code, char a2)
 	{
-		LOG(WARNING) << "Error Code: " <<
-			string_conversions::utf_16_to_code_page(
-				CP_UTF8,
-				g_pointers->m_gta.m_get_title_caption_error_message_box(joaated_error_code));
+		if (joaated_error_code == 0x971454B7)
```
Comment (MarkEcza) [inline src/hooks/misc/log_message_error_box.cpp]: Can you comment "ERR_NET_EVENT" to clarify or maybe JOAAT("ERR_NET_EVENT")

File: src/hooks/protections/allocate_memory_reliable.cpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+		while (cxn->m_normal_message_queue.m_count)
+		{
+			auto msg = cxn->m_normal_message_queue.m_first;
+
+			g_pointers->m_gta.m_remove_message_from_queue(&cxn->m_normal_message_queue, msg);
+
+			if (is_reliable_message(msg))
+				g_pointers->m_gta.m_remove_message_from_unacked_reliables(&cxn->m_unacked_reliable_message_list, &msg->word4C);
+			else
+				LOG(INFO) << "hmm";
```
Comment (Yimura) [inline src/hooks/protections/allocate_memory_reliable.cpp]: Still intended to be left here?

**Review by Yimura — CHANGES_REQUESTED** [review]

File: src/hooks/protections/allocate_memory_reliable.cpp (hunk 2 above)
Comment (gir489returns) [reply src/hooks/protections/allocate_memory_reliable.cpp:56]: image

File: src/hooks/misc/log_message_error_box.cpp (hunk 1 above)
Comment (maybegreat48) [reply src/hooks/misc/log_message_error_box.cpp]: Thanks

**Review by Yimura — DISMISSED** [review]

**Review by Yimura — APPROVED** [review]

**Pull request merged**